File: NUnit/Version2/Saucery2/Registry/RegistryHelper.cs
using System.Security.Principal;
using Saucery2.Util;

namespace Saucery2.Registry {
    internal class RegistryHelper {
        readonly static ModifyRegistry RegModifier = new ModifyRegistry(IsAdministrator());

        public static bool IsAlreadyActivated() {
            #if DEBUG
                var key = RegModifier.Read(SauceryConstants.KEYNAME);
                return key != null && (!key.Equals(string.Empty) && !string.IsNullOrEmpty(key));
                //return true;
            #else
                var key = RegModifier.Read(SauceryConstants.KEYNAME);
                return key != null && (!key.Equals(string.Empty) && !string.IsNullOrEmpty(key));
#endif
        }

        public static void StoreActivation(string publicKey) {
            RegModifier.Write(SauceryConstants.KEYNAME, publicKey);
        }

        public static bool IsAdministrator() {
            return new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator);
        }
    }
}
/*
 * Author: Andrew Gray, Full Circle Solutions
 * Date: 22nd September 2010
 * 
 */

File: JUnit/saucery-core/src/main/java/au/com/fullcirclesolutions/saucery/registry/RegistryHelper.java
package au.com.fullcirclesolutions.saucery.registry;

import java.util.prefs.Preferences;

import au.com.fullcirclesolutions.saucery.utils.SauceryConstants;

public class RegistryHelper {

    static ModifyRegistry RegModifier = new ModifyRegistry(IsAdministrator());

    public static boolean IsAlreadyActivated() {
        String key = RegModifier.Read(SauceryConstants.KEYNAME);
        return key != null && (!key.equals(SauceryConstants.EMPTY_STRING));
    }

    public static void StoreActivation(String publicKey) {
        RegModifier.Write(SauceryConstants.KEYNAME, publicKey);
    }

    public static boolean IsAdministrator() {
        Preferences prefs = Preferences.systemRoot();
        try {
            prefs.put("foo", "bar"); // SecurityException on Windows
            prefs.remove("foo");
            prefs.flush(); // BackingStoreException on Linux
            return true;
        } catch (Exception e) {
            return false;
        }
    }
}
/*
 * Copyright: Andrew Gray, Full Circle Solutions
 * Date: 22nd September 2010
 */